frontend/src/util/Hook.tsx

```diff
@@ -30,6 +30,52 @@ export const useBestSubmission = (player?: Player | null) => {
   return bestSubmission;
 };
 
+export const useGetScore = (player?: Player) => {
+  const counted = new Set<number>();
+  const [score, setScore] = useState<number>(0);
+
+  useEffect(() => {
+    if (player) {
+      for (let i = 0; i < player.submissions.length; i += 1) {
+        if (player.submissions[i].numCorrect === player.submissions[i].numTestCases
+          && !counted.has(player.submissions[i].problemIndex)) {
+          counted.add(player.submissions[i].problemIndex);
+        }
+      }
+
+      setScore(counted.size);
+    }
+  }, [player, setScore, counted]);
+
+  if (player == null || player.submissions.length === 0) {
+    return null;
+  }
+  return score;
+};
+
+export const useGetSubmissionTime = (player?: Player) => {
```

Comment:
Yeah, so the reason I didn't go backwards was because if the user submitted the same correct code twice, useGetSubmissionTIme would get the time of the second one. I believe we agreed that we want the first submission that solves the last solved problem.

Reply:
You're right - I realized this later but forgot to remove this comment. Thanks for thinking of that! 💯